pandas/tests/indexes/ranges/test_range.py

```diff
@@ -295,6 +295,24 @@ def test_cache(self):
         expected = np.arange(0, 100, 10, dtype="int64")
         tm.assert_numpy_array_equal(idx._cache["_data"], expected)
 
+    def test_cache_after_calling_loc_with_array(self):
+        # GH 53387
+        # the cache will contain a _constructor key, so it should be tested separately
+        idx = RangeIndex(0, 100, 10)
+        df = pd.DataFrame({"a": range(10)}, index=idx)
+
+        assert "_data" not in idx._cache
+
+        # take is internally called by loc, but it's also tested explicitly
+        idx.take([3, 0, 1])
+        assert "_data" not in idx._cache
+
+        df.loc[[50]]
+        assert "_data" not in idx._cache
```

Comment:
This new test is failing on windows and linux-32-bit [here](https://github.com/pandas-dev/pandas/actions/runs/5089852637/jobs/9147968483?pr=53397#step:5:331) because `_data` is initialized even though it shouldn't.

On linux-32-bit, it's because:
- this if-branch is entered: https://github.com/pandas-dev/pandas/blob/8cf4ab491b06274f555d1ea305c172354208747a/pandas/core/indexes/base.py#L3915
- `_find_common_type_compat` is called, and `self._values` is accessed here: https://github.com/pandas-dev/pandas/blob/8cf4ab491b06274f555d1ea305c172354208747a/pandas/core/indexes/base.py#L6248

I think it can be fixed by changing the call to `find_result_type` with a call to `find_common_type` (that relies on dtype, without accessing values), but it may need some additional work to be successful with IntervalIndex.

If this doesn't work, I can just overwrite the method `_find_common_type_compat` in the `RangeIndex` class.

Reply:
How hard would it be to change find_result_type to take a dtype instead of values?  If that isn't too difficult, that would be a better solution.

Another long-term solution would be to implement a RangeArray to back RangeIndex, but thats definitely out of scope here.

Reply:
Done, changed find_result_type to take a dtype instead of values.
Since the function is used in just a couple of places, it didn't need too many changes.

Reply:
Nope.  Looks like I've mentioned the idea in #37277 and #19954 but never fleshed it out

Reply:
Yes, because now `find_result_type` has been modified to work with the dtype, and when it's called from `_find_common_type_compat` it doesn't load `_data` anymore.